spec/models/tag_spec.rb

```diff
@@ -92,6 +92,11 @@
       end
     end
 
+    it "fails validation if name is a prohibited (whitespace) unicode character" do
+      tag.name = "U+202D"
+      expect(tag).not_to be_valid
```

Comment:
it's possible this unit passes (tag is invalid) for reasons not related to the intent you had, I think (because of the way ruby represents characters in strings) you want something like this instead?

```ruby
  tag.name = "\u202d" 
  expect(tag).not_to be_valid
  expect(tag.errors).to include(a specific message expected when tag contains invalid whitespace)
```